packages/assets-controllers/src/NftDetectionController.ts

```diff
@@ -794,12 +799,20 @@ export class NftDetectionController extends BaseController<
               collection && { collection },
               chainId && { chainId },
             );
-            await this.#addNft(contract, tokenId, {
-              nftMetadata,
-              userAddress,
-              source: Source.Detected,
-              chainId: toHex(chainId),
-            });
+            const networkClientId = this.messagingSystem.call(
+              'NetworkController:findNetworkClientIdByChainId',
+              toHex(chainId as number),
+            );
+            await this.#addNft(
+              contract,
+              tokenId,
+              networkClientId as NetworkClientId,
```

Comment:
Why the type assertion here?

Reply:
Thanks! Fixed here [42b9206](https://github.com/MetaMask/core/pull/5622/commits/42b9206dbeca78be6b46a7e798425d155954e790)